### PR TITLE
Add more lint coverage

### PR DESCRIPTION
**Pull request by rosstimothy**
`golangci-lint` doesn't lint any subdirectories with their own `go.mod` (https://github.com/golangci/golangci-lint/issues/828) which left a few directories unlinted. To get around this we can
run `golangci-lint` directly against those submodules.

This also provides fixes for new lint errors detected by running `golangci-lint`
in these directories.

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -102,8 +102,8 @@ ENV GOPATH="/go" \
 RUN go install github.com/google/addlicense@v1.0.0
 
 # Install meta-linter.
-RUN (curl -L https://github.com/golangci/golangci-lint/releases/download/v1.38.0/golangci-lint-1.38.0-$(go env GOOS)-$(go env GOARCH).tar.gz | tar -xz && \
-     cp golangci-lint-1.38.0-$(go env GOOS)-$(go env GOARCH)/golangci-lint /bin/ && \
+RUN (curl -L https://github.com/golangci/golangci-lint/releases/download/v1.44.0/golangci-lint-1.44.0-$(go env GOOS)-$(go env GOARCH).tar.gz | tar -xz && \
```
Comment (tcsc) [inline build.assets/Dockerfile:105]: We will have to re-spin the buildbox, or update the linter script, in order for this update to make it onto CI.

Comment (rosstimothy) [reply build.assets/Dockerfile:105]: Yeah I figured there would be more involved than this. But I was constantly getting errors that aren't picked up by CI and believe it is due to the older version.

Comment (tcsc) [reply build.assets/Dockerfile:108]: I have PR #10179 that will remove some of the manual buildbox image management for GCB

**Review by tcsc — APPROVED** [review]

File: build.assets/Dockerfile (hunk 1 above)
Comment (zmb3) [reply build.assets/Dockerfile:108]: nit: I think the comment here is out of date. If I had to guess, at some point we switched from `go-metalinter` to `golangci-lint` and left the old install comment.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**